src/native/external/zlib-ng.cmake

```diff
@@ -15,12 +15,14 @@ add_compile_options($<$<COMPILE_LANG_AND_ID:C,MSVC>:/wd4127>) # warning C4127: c
 add_compile_options($<$<COMPILE_LANG_AND_ID:C,MSVC>:/wd4242>) # 'function': conversion from 'unsigned int' to 'Pos', possible loss of data, in various deflate_*.c files
 add_compile_options($<$<COMPILE_LANG_AND_ID:C,MSVC>:/wd4244>) # 'function': conversion from 'unsigned int' to 'Pos', possible loss of data, in various deflate_*.c files
 
+# NOTE: Wasm is temporarily using system zlib, but will be switched to zlib-ng eventually. This branch should not affect the build.
 # 'aligned_alloc' is not available in browser/wasi, yet it is set by zlib-ng/CMakeLists.txt.
 if (CLR_CMAKE_TARGET_BROWSER OR CLR_CMAKE_TARGET_WASI)
   set(HAVE_ALIGNED_ALLOC FALSE CACHE BOOL "have aligned_alloc" FORCE)
 endif()
 
-set(BUILD_SHARED_LIBS OFF) # Shared libraries aren't supported in wasm
+set(BUILD_SHARED_LIBS OFF) # Shared libraries aren't supported in wasm, and even though it consumes system zlib, it will be switched to zlib-ng eventually.
```

Comment:
Since we plan on adding zlib-ng support to wasm _eventually_ (this is just temporary), I decided to not remove this line that disables shared libs, otherwise I would have to make a ton of changes to the PR.